vcd/resource_vcd_edgegateway_settings.go

```diff
@@ -0,0 +1,194 @@
+				Optional:    true,
+				Description: "Enable load balancing. (Disabled by default)",
+			},
+			"lb_acceleration_enabled": &schema.Schema{
+				Type:        schema.TypeBool,
+				Default:     false,
+				Optional:    true,
+				Description: "Enable load balancer acceleration. (Disabled by default)",
+			},
+			"lb_logging_enabled": &schema.Schema{
+				Type:        schema.TypeBool,
+				Default:     false,
+				Optional:    true,
+				Description: "Enable load balancer logging. (Disabled by default)",
+				// Due to a vCD bug, this field can only be changed by a system administrator
+			},
+			"lb_loglevel": &schema.Schema{
+				Type:         schema.TypeString,
+				Default:      "info",
+				Optional:     true,
+				ValidateFunc: validateCase("lower"),
+				Description: "Log level. One of 'emergency', 'alert', 'critical', 'error', " +
+					"'warning', 'notice', 'info', 'debug'. ('info' by default)",
+				// Due to a vCD bug, this field can only be changed by a system administrator
+			},
+			"fw_enabled": &schema.Schema{
+				Type:        schema.TypeBool,
+				Default:     true,
+				Optional:    true,
+				Description: "Enable firewall. Default 'true'",
+			},
+			"fw_default_rule_logging_enabled": &schema.Schema{
+				Type:        schema.TypeBool,
+				Default:     false,
+				Optional:    true,
+				Description: "Enable logging for default rule. Default 'false'",
+			},
+			"fw_default_rule_action": &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "deny",
+				Description:  "'accept' or 'deny'. Default 'deny'",
+				ValidateFunc: validation.StringInSlice([]string{"accept", "deny"}, false),
+			},
+		},
+	}
+}
+
+func resourceVcdEdgeGatewaySettingsCreate(d *schema.ResourceData, meta interface{}) error {
+	return resourceVcdEdgeGatewaySettingsUpdate(d, meta)
+}
+
+func getVcdEdgeGateway(d *schema.ResourceData, meta interface{}) (*govcd.EdgeGateway, error) {
+
+	log.Printf("[TRACE] edge gateway settings read initiated")
+
+	vcdClient := meta.(*VCDClient)
+
+	orgName := d.Get("org").(string)
+	vdcName := d.Get("vdc").(string)
+	_, vdc, err := vcdClient.GetOrgAndVdc(orgName, vdcName)
+	if err != nil {
+		return nil, fmt.Errorf("[getVcdEdgeGateway] error retrieving org and vdc: %s", err)
+	}
+	var edgeGateway *govcd.EdgeGateway
+
+	// Preferred identification method is by ID
+	identifier := d.Get("edge_gateway_id").(string)
+	if identifier == "" {
+		// Alternative method is by name
+		identifier = d.Get("name").(string)
+	}
+	if identifier == "" {
+		return nil, fmt.Errorf("[getVcdEdgeGateway] no identifier provided")
+	}
+	edgeGateway, err = vdc.GetEdgeGatewayByNameOrId(identifier, false)
+	if err != nil {
+		return nil, fmt.Errorf("[getVcdEdgeGateway] edge gateway %s not found: %s", identifier, err)
+	}
+	return edgeGateway, nil
+}
+
+func resourceVcdEdgeGatewaySettingsRead(d *schema.ResourceData, meta interface{}) error {
+	edgeGateway, err := getVcdEdgeGateway(d, meta)
+	if err != nil {
+
```

Comment:
not needed empty line

Reply:
Thanks. Removed. 
Also added import function that was left behind.